src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public Figure getRandomFigure() {
+        Figure randomFigure;
+        int randomIndex = new Random().nextInt(5);
+        switch (randomIndex) {
+            case 0 -> randomFigure = getRandomCircle();
+            case 1 -> randomFigure = getIsoscelesTrapezoid();
+            case 2 -> randomFigure = getRandomRectangle();
+            case 3 -> randomFigure = getRightTriangle();
+            case 4 -> randomFigure = getRandomSquare();
+            default -> randomFigure = new Figure();
```

Comment:
The default case in the switch statement creates a new `Figure` object. If `Figure` is intended to be an abstract class or should not be instantiated directly, consider removing this default case or handling it differently.